### PR TITLE
FUSETOOLS2-1178 - include new Option names for modeline

### PR DESCRIPTION
**Pull request by apupier**
Screenshot from 2021-06-25 14-48-56
Screenshot from 2021-06-25 14-48-50

### CODE REVIEW

**Review by bfitzpat — APPROVED** [review]

File: src/main/java/com/github/cameltooling/lsp/internal/completion/modeline/CamelKModelineOptionNames.java [hunk 1]
```diff
@@ -38,8 +38,17 @@ private CamelKModelineOptionNames() {}
...
 	
+	private static final String DESCRIPTION_SINCE_COMMUNITY_1_5_PROD_1_4 = "Option is available since Community 1.5 and Red Hat productized 1.4";
+	
 	static {
 		OPTION_NAMES_WITH_DESCRIPTION = new HashMap<>();
+		OPTION_NAMES_WITH_DESCRIPTION.put("build-property",
+				"Add a build time property or properties file (syntax: [my-key=my-value|file:/path/to/my-conf.properties]\n\n"
+				+ DESCRIPTION_SINCE_COMMUNITY_1_5_PROD_1_4);
+		OPTION_NAMES_WITH_DESCRIPTION.put("config",
+				"Add a runtime configuration from a Configmap, a Secret or a file (syntax: [configmap|secret|file]:name[/key],"
```
Comment (bfitzpat) [inline src/main/java/com/github/cameltooling/lsp/internal/completion/modeline/CamelKModelineOptionNames.java:49]: Might change the text slightly to remove the unnecessary indefinite articles in the list... "Add a runtime configuration from a ConfigMap, Secret, or file..."

Comment (apupier) [reply src/main/java/com/github/cameltooling/lsp/internal/completion/modeline/CamelKModelineOptionNames.java:49]: this text is duplicated from upstream https://camel.apache.org/camel-k/latest/cli/modeline.html I will provide the suggestion and if accepted, update here to have it in phase

Comment (apupier) [reply src/main/java/com/github/cameltooling/lsp/internal/completion/modeline/CamelKModelineOptionNames.java:49]: https://github.com/apache/camel-k/pull/2451

Comment (apupier) [reply src/main/java/com/github/cameltooling/lsp/internal/completion/modeline/CamelKModelineOptionNames.java]: merged in master upstream.
updated PR

**Review by lhein — APPROVED** [review]
It would be really nice if these options and their descriptions would be provided and not hard coded.

**Comment by apupier** [comment]
> It would be really nice if these options and their descriptions would be provided and not hard coded.

agree. Reported https://issues.redhat.com/browse/FUSETOOLS2-1184

**Pull request merged**